### PR TITLE
Character physics refactoring

### PR DESCRIPTION
**Pull request by aryasalehi**
Related : https://github.com/webaverse/app/pull/3591#pullrequestreview-1071987019

### CODE REVIEW

File: avatars/avatars.js [hunk 1]
```diff
@@ -641,6 +641,7 @@ class Avatar {
 
     // height is defined as eyes to root
     this.height = getHeight(object);
+    this.width = 0.36; // TODO : calculate this instead of hard coding it
```
Comment (avaer) [inline avatars/avatars.js:644]: 👍

**Review by avaer — APPROVED** [review]
LGTM!

**Comment by avaer** [comment]
A proposal for further refactoring, LMK what you think:


- Rename `Player`/`PlayerBase` -> Character`. This has all of the actions, high level stats, and auxiliary classes for things like sounds and special effects.
- A `Character` of a VRM (previously called `Player`) is an `AvatarCharacter`
- Rename `Avatar` -> `AvatarAnimator` since its job is to apply `Player` actions to animate an avatar.
- Introduce a new GLB (non-VRM) `Character` called a `ModelCharacter` which is similar to an `AvatarCharacter`
- A new `ModelAnimator` class which serves the same role as `AvatarAnimator` (previously called `Avatar`)
- A mob is a `MobCharacter` with a `ModelAnimator`
- A pet is a `PetCharacter` with a `ModelAnimator`

**Comment by avaer** [comment]
Then it would look like:

```
AvatarCharacter -> {
  CharacterPhysics -> PhysX character controller,
  CharacterSfx,
  AvatarAnimator -> AvatarRenderer -> VRM totum app,
}
MobCharacter -> {
  CharacterPhysics -> PhysX character controller,
  CharacterSfx,
  ModelAnimator -> GLB totum app,
}
PetCharacter -> {
  CharacterPhysics -> PhysX character controller,
  CharacterSfx,
  ModelAnimator -> GLB totum app,
}
```

**Comment by aryasalehi** [comment]
> Then it would look like:
> 
> ```
> AvatarCharacter -> {
>   CharacterPhysics -> PhysX character controller,
>   CharacterSfx,
>   ...,
>   AvatarAnimator -> AvatarRenderer -> VRM totum app,
> }
> MobCharacter -> {
>   CharacterPhysics -> PhysX character controller,
>   CharacterSfx,
>   ...,
>   ModelAnimator -> GLB totum app,
> }
> PetCharacter -> {
>   CharacterPhysics -> PhysX character controller,
>   CharacterSfx,
>   ...,
>   ModelAnimator -> GLB totum app,
> }
> ```
I think this makes sense 👌👌

**Pull request merged**